pkg/components/component.go

```diff
@@ -39,13 +39,13 @@ func NewTestIdentifier(reg *registry.Registry, componentIDs map[string]int64) *T
 	}
 }
 
-func (t *TestIdentifier) Identify(test *v1.TestInfo) (*v1.TestOwnership, error) {
+func (t *TestIdentifier) Identify(test v1.TestInfo) (*v1.TestOwnership, error) {
```

Comment:
What's the reasoning behind all the changes to make TestInfo passed by value instead of reference? Doesn't seem consistently done (e.g. IdentifyTest)

Using a pointer lets you detect missing inputs via nil, acting as a guardrail. With a value, zero fields may go unnoticed and cause bugs

Reply:
I didn't want to update the reference passed in within the `setDefaults` method, so I thought that passing it by value would be the simplest way to not do that. I can change the way it works though and introduce a new variable instead.